tests/functional/test_s3.py

```diff
@@ -3487,6 +3487,38 @@ def _addressing_for_presigned_url_test_cases():
         signature_version="s3",
         expected_url="https://s3.us-west-1.amazonaws.com/foo.bar.biz/key",
     )
+    # Bucket names that contain dots and subcomponents that are less than
+    # 3 characters should still use virtual host style addressing if
+    # configured by the customer and they provide their own ``endpoint_url``
+    # that is insecure. https://github.com/boto/botocore/issues/2938
+    yield dict(
+        bucket="foo.b.biz",
+        key="key",
+        s3_config={"addressing_style": "virtual"},
+        customer_provided_endpoint="http://s3.us-west-2.amazonaws.com",
+        expected_url="http://foo.b.biz.s3.us-west-2.amazonaws.com/key",
+    )
+    yield dict(
+        bucket="foo.b.biz",
+        key="key",
+        s3_config={"addressing_style": "virtual"},
+        is_secure=False,
+        expected_url="http://s3.amazonaws.com/foo.b.biz/key",
+    )
+    yield dict(
+        bucket="foo.b.biz",
+        key="key",
+        s3_config={"addressing_style": "virtual"},
+        customer_provided_endpoint="https://s3.us-west-2.amazonaws.com",
+        expected_url="https://s3.us-west-2.amazonaws.com/foo.b.biz/key",
+    )
+    yield dict(
+        region="us-west-1",
+        bucket="foo.b.biz",
+        key="key",
+        s3_config={"addressing_style": "virtual"},
+        expected_url="https://s3.us-west-1.amazonaws.com/foo.b.biz/key",
+    )
```

Comment:
What are we testing between these two? Region shouldn't be related to virtual construction unless we're looking at us-east-1 which I think is already covered above.

Reply:
Sure. I'll remove.